changes/256.bugfix.rst

```diff
@@ -0,0 +1,2 @@
+Retain Objective-C objects when creating Python wrappers and release them when the
+Python wrapped is garbage collected.
```

Comment:
> This change should be backward compatible for most users because existing manual retain and release calls don't cause any issues if balanced

Unfortunately, as you've discovered, existing calls may not be balanced if they depended on the previous `_needs_release` behavior. So we should increase the version number to 0.5.0 to prevent older Toga versions from seeing the new behavior, and make this a "removal" change note with advice on how to adapt to it.

And if we get this right, that advice should simply be "remove ALL instances of retain, release and autorelease from your code, and never worry about it again". If we can do that in Toga, then there's a pretty good chance that it will work for other clients as well.

Reply:
Indeed, this will require migration advice in the release notes. Will add that.

Regarding the new version number, how does this work? Just tag a commit? You seam to have something fancy going on these days with setuptools_scm.

Reply:
Yes, nothing is required in the PR itself; we would tag the main branch after it's merged.

Reply:
Yes, nothing is required in the PR itself; we would tag the main branch after it's merged.